### PR TITLE
[KOGITO-3105] - Missing status for failures for KogitoRuntime CRs

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submitting your Pull Request :heart:! 

Jira : https://issues.redhat.com/browse/KOGITO-3105
With this fix, conditions will looks as following :
```
Status:
  Cloud Events:
  Conditions:
    Last Transition Time:  2021-03-25T09:19:17Z
    Message:               you may not have access to the container image "quay.io/kiegroup/process-quarkus-example-1:latest"
    Reason:                ReconciliationFailure
    Status:                True
    Type:                  Failed
    Last Transition Time:  2021-03-25T09:19:17Z
    Message:               
    Reason:                RequestedReplicasNotEqualToAvailableReplicas
    Status:                False
    Type:                  Provisioning
    Last Transition Time:  2021-03-25T09:19:17Z
    Message:               
    Reason:                NoPodAvailable
    Status:                False
    Type:                  Deployed

```

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: controllers/kogitoruntime_controller_test.go [hunk 1]
```diff
@@ -145,3 +148,59 @@ func TestReconcileKogitoRuntime_CustomConfigMap(t *testing.T) {
...
+	cli := test.NewFakeClientBuilder().AddK8sObjects(instance, imageStream).OnOpenShift().Build()
+	r := &KogitoRuntimeReconciler{Client: cli, Scheme: meta.GetRegisteredSchema(), Log: test.TestLogger}
+	_, err := r.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}})
+	assert.Error(t, err)
+
+	_, err = kubernetes.ResourceC(cli).Fetch(instance)
+	assert.NoError(t, err)
+	assert.NotNil(t, instance.Status)
+	assert.Len(t, *instance.Status.Conditions, 3)
+	failedCondition := meta2.FindStatusCondition(*instance.Status.Conditions, string(api.FailedConditionType))
```
Comment (ricardozanini) [inline controllers/kogitoruntime_controller_test.go:203]: I like this!

**Review by ricardozanini — APPROVED** [review]
Looks neat!

File: controllers/kogitoruntime_controller_test.go (hunk 1 above)
Comment (vaibhavjainwiz) [reply controllers/kogitoruntime_controller_test.go:203]: :)

**Review by spolti — APPROVED** [review]

**Review by Kaitou786 — APPROVED** [review]

**Comment by sutaakar** [comment]
The error is properly reflected in status.
But I have noticed one thing, such error triggers reconciliation every second. IMHO we should define longer reconciliation status to avoid spamming operator. Maybe once in a minute?

**Comment by ricardozanini** [comment]
> The error is properly reflected in status.
> But I have noticed one thing, such error triggers reconciliation every second. IMHO we should define longer reconciliation status to avoid spamming operator. Maybe once in a minute?

Agreed. The reconciliation cannot start immediately after an error like this. One minute seems reasonable.

**Comment by vaibhavjainwiz** [comment]
I set the reconcilation time to 1 min if any error occur while resolving image. Now operator will not go mad. Conditions will looks like following :
```
  Conditions:
    Last Transition Time:  2021-03-26T16:07:14Z
    Message:               you may not have access to the container image "quay.io/kiegroup/process-quarkus-example-1:latest"
    Reason:                ImageStreamNotReadyReason
    Status:                True
    Type:                  Failed
    Last Transition Time:  2021-03-26T16:07:14Z
    Message:               
    Reason:                RequestedReplicasEqualToAvailableReplicas
    Status:                True
    Type:                  Provisioning
    Last Transition Time:  2021-03-26T16:07:14Z
    Message:               
    Reason:                NoPodAvailable
    Status:                False
    Type:                  Deployed
```

**Review by sutaakar — APPROVED** [review]

**Pull request merged**